server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -222,8 +222,14 @@ export class TenantManager {
             customData.externalStorageData.accessInfo = this.encryptAccessInfo(accessInfo);
         }
         await collection.update({ _id: tenantId }, { customData }, null);
-
-        return (await this.getTenantDocument(tenantId)).customData;
+        const tenantDocument = await this.getTenantDocument(tenantId, true);
```

Comment:
Current behavior(without this change) is that it will always update the tenant's custom data, whether the tenant is disabled or not. For active tenant, it return success but for disabled tenant, it returns 400 just because "(await this.getTenantDocument(tenantId))" returns null, which throws an error (cannot get customData of null). That means the update does happen even if the caller receives the error response. With this change, I think caller should receive success if update success, otherwise it gets the wrong impression. And we don't have many scenario to update a disabled tenant.

For you suggestion, I don't think it is necessary since usually the caller will use TenantExist or getTenant to check tenant existence and both methods have includeDisabledTenant flag. When it comes to the update, the caller should already knows what it needs to do.

Reply:
Makes sense! Thanks for providing the details!